File: Mediator/Ruby/LockingSystem.rb
class LockingSystem
    def initialize
        @status = false
        @command = nil
    end

    def lock(command)
        @command = command
        @status = true
    end

    def unlock
        @status = false
    end

    def checkStatus
        if @status
            puts "The system is currently used by:"
            @command.describe
        end

        return @status
    end
end

File: Mediator/Dart/LockingSystem.dart
import 'Command.dart';

class LockingSystem {
  bool? _status;
  Command? _command;

  LockingSystem() {
    _status = false;
  }

  lock(Command command) {
    _command = command;
    _status = true;
  }

  unlock() {
    _status = false;
  }

  bool checkStatus() {
    if (_status == true) {
      print("The system currently used by: ");
      _command?.describe();
    }

    return _status ?? false;
  }
}
